exdb/tests.py

```diff
@@ -252,3 +258,124 @@ def test_does_not_get_spontaneous(self):
+        form = ExperienceSubmitForm(data, when=self.test_date)
+        self.assertFalse(form.is_valid(), "Form should NOT have been valid")
+
+    def test_past_experience_creation_no_attendance(self):
+        self.create_type(needs_verification=False)
+        data = {'name': 'test', 'description': 'test', 'start_datetime': (self.test_date - timedelta(days=2)),
+                'end_datetime': (self.test_date - timedelta(days=1)), 'type': 2, 'sub_type': 1, 'audience': 'c',
+                'guest': '1', 'recognition': [1], 'keywords': [1], 'goal': 'a'}
+        form = ExperienceSubmitForm(data, when=self.test_date)
+        self.assertFalse(form.is_valid(), "Form should NOT have been valid")
+
+    def test_experience_creation_with_attendance(self):
+        data = {'name': 'test', 'description': 'test', 'start_datetime': (self.test_date + timedelta(days=1)),
+                'end_datetime': (self.test_date + timedelta(days=2)), 'type': 1, 'sub_type': 1, 'audience': 'c',
+                'guest': '1', 'recognition': [1], 'keywords': [1], 'goal': 'a', 'attendance': 1}
+        form = ExperienceSubmitForm(data, when=self.test_date)
+        self.assertFalse(form.is_valid(), "Form should NOT have been valid")
+
+    def test_past_experience_creation_negative_attendance(self):
+        self.create_type(needs_verification=False)
+        data = {'name': 'test', 'description': 'test', 'start_datetime': (self.test_date - timedelta(days=2)),
+                'end_datetime': (self.test_date - timedelta(days=1)), 'type': 2, 'sub_type': 1, 'audience': 'c',
+                'guest': '1', 'recognition': [1], 'keywords': [1], 'goal': 'a', 'attendance': -1}
+        form = ExperienceSubmitForm(data, when=self.test_date)
+        self.assertFalse(form.is_valid(), "Form should NOT have been valid")
+
+
+class ExperienceCreationViewTest(StandardTestCase):
+    def setUp(self):
+        self.test_user = get_user_model().objects.create_user('test_user', 't@u.com', 'a')
+        self.test_type = Type.objects.create(name="Test Type", needs_verification=True)
+        self.test_sub_type = SubType.objects.create(name="Test Sub Type")
+        self.test_org = Organization.objects.create(name="Test Organization")
+        Keyword.objects.create(name="test keyword")
+
+    def test_valid_future_experience_creation_view_submit(self):
+        c = Client()
+        start = now() + timedelta(days=1)
+        end = now() + timedelta(days=2)
+        c.login(username="test_user", password='a')
+        data = {'name': 'test', 'description': 'test', 'start_datetime_month': start.month,
+                'start_datetime_day': start.day, 'start_datetime_year': start.year,
+                'end_datetime_month': end.month, 'end_datetime_day': end.day, 'end_datetime_year': end.year,
+                'type': 1, 'sub_type': 1, 'audience': 'c',
+                'guest': '1', 'recognition': 1, 'keywords': 1, 'goal': 'a', 'submit': 'Submit'}
```

Comment:
All the hard-coded pks cause the tests to fail when using postgres as the database.